apps/web/app/admin/members/_components/SidePanel.tsx

```diff
@@ -32,22 +41,76 @@ const initialFormData: SidePanelFormData = {
 };
 
 export default function SidePanel({ isOpen, onClose, selectedMember }: AddMemberSidePanelProps): JSX.Element {
-  const [formData, setFormData] = useState<SidePanelFormData>(initialFormData);
+  const queryClient = useQueryClient();
   const [imageObjectUrl, setImageObjectUrl] = useState<string>("");
   const [isImageError, setIsImageError] = useState(false);
+  const sidePanelRef = useRef<HTMLDivElement>(null);
 
-  const handleRoleChange = (role: string): void => {
-    setFormData((prev) => ({ ...prev, role: role as IUser["role"] }));
-  };
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    control,
+    setValue,
+    watch,
+    reset,
+  } = useForm<SidePanelFormData>({
+    defaultValues: initialFormData,
+  });
 
-  const handleInputChange = (e: ChangeEvent<HTMLInputElement>): void => {
-    const { id, value } = e.target;
-    setFormData((prev) => ({ ...prev, [id]: value }));
-  };
+  const postMemberMutation = useMutation({
+    mutationFn: postMember,
+    onSuccess: async () => {
+      notify({
+        type: "success",
+        message: selectedMember ? NOTIFICATION_MESSAGES.MEMBER_UPDATE : NOTIFICATION_MESSAGES.MEMBER_ADD,
+      });
+      await queryClient.invalidateQueries({ queryKey: ["members"] });
+      onClose();
+    },
+    onError: () => {
+      notify({
+        type: "error",
+        message: "멤버 추가에 실패했습니다.",
+      });
+    },
+  });
 
-  const handleTeamsSelect = (teams: string[]): void => {
-    setFormData((prev) => ({ ...prev, teams }));
-  };
+  const patchMemberMutation = useMutation({
+    mutationFn: (data: FormData) => (selectedMember ? patchMember(selectedMember._id, data) : postMember(data)),
+    onSuccess: async () => {
+      notify({
+        type: "success",
+        message: selectedMember ? NOTIFICATION_MESSAGES.MEMBER_UPDATE : NOTIFICATION_MESSAGES.MEMBER_ADD,
+      });
+      await queryClient.invalidateQueries({ queryKey: ["members"] });
+      onClose();
```

Comment:
```suggestion
      queryClient.invalidateQueries({ queryKey: ["members"] });
      onClose();
```
여기도 혹시 async/await없이 동작하는지만 확인 부탁할게!